### DISCUSSION TITLE
Welcome to bash-oo-framework Discussions!

### DISCUSSION THREAD

**Discussion by niieani** [niieani/bash-oo-framework#89] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by adudek** [comment]
Since next release will have garbage collector anyways, lets ditch concept of explicit includes altogether and make writing code "smart". Just like bash expects your binaries be in $PATH so should libs/types be.

**Reply by dzove855** [reply]
So your idea is to put libs/types in $PATH if i see it right?

I would prefere something different, like creating a new var BASH_LIB and put it there with the same idea how PATH works, like this the user have his own way to do it in example, if someone would like to put libs in $PATH he can just do, this:
BASH_LIB=$PATH and that should work.

Also i would consider the possiblity that bash provides loadable builtins, which can help a lot writing own parser for json etc.. (https://mywiki.wooledge.org/BashLoadableBuiltins).

On the other side, i think the way how this framework handles multiple version i would prefer something different to keep the code clean, something like splitting the libs per version, yes it will be more work, but on the other side, if someone just wants to be bash5 compliant he don't need any kind of 'hack' in his libs and he will keep a clean code

**Reply by adudek** [reply]
Yep, your BASH_LIB proposition is exatly what was in my mind, also its not so hard to "compile" proper BASH_LIB value based on `bash --version`. Thing is that without docker it will be darn hard to debug other people's code and with that it still needs to be limited to sane number of variances.

**Reply by dzove855** [reply]
What actually could help, instead of having some kind of docker environment, is to implement a dependency manager. 
I think some already exist but not sure it will fits the needs. 
Just a simple depends file with links inside and the manager will load every library needed in some kind of vendor/libs also if some lib have dependencies. 
This can help to simulate a user environment. 

The idea of this is, that it will be simple to write plugins around it without to have some big messy framework.
